packet/flatcar-linux/kubernetes/calico/host-protection.yaml.tmpl

```diff
@@ -1,126 +1,109 @@
-# Customize inbound failsafe rules
-# https://docs.projectcalico.org/v3.5/getting-started/bare-metal/policy/failsafe
+# General
+# =======
+---
+# Disable default failsafe inbound rules
+apiVersion: crd.projectcalico.org/v1
+kind: FelixConfiguration
+metadata:
+  name: default
+spec:
+  failsafeInboundHostPorts: []
+
+# Host endpoints
+# ==============
+---
+${controller_host_endpoints}
+---
+${worker_host_endpoints}
+
+# Firewall policy
+# ===============
+---
+# SSH access
 apiVersion: crd.projectcalico.org/v1
 kind: GlobalNetworkPolicy
 metadata:
-  name: failsafe-ingress
+  name: ssh
 spec:
+  selector: has(host-endpoint)
```

Comment:
Hm, can't we use default value `all()` here?

Reply:
According to the [docs](https://docs.projectcalico.org/v3.7/reference/calicoctl/resources/globalnetworkpolicy#selector) `has()` is more specific than `all()`, so we would be loosening the selector by switching.

Reply:
What do you mean by `loosening the selector by switching`? That it will catch potentially more resources than it should? Do we plan to have more resources with different label? Or is it a good practice to always use some label selector? I usually prefer to rely on upstream defaults, so there is less "custom" configuration you need to read and understand.

Reply:
I **think** we want `has(host-endpoint)` because all applies to all calico resources, not just host-endpoint. Therefore, if the user uses calico for limiting inter-pod communication, this rule will apply for the resources created for that.

But, I'm new to calico, I might be wrong :)

Reply:
>GlobalNetworkPolicy applies to workload endpoint resources in all namespaces, and to host endpoint resources.

I wasn't aware that `GlobalNetworkPolicy` also applies to `WorkloadEndpoint` resource (I didn't even know it exists), all good then :+1: